tests/src/core/testqgsexpression.cpp

```diff
@@ -1246,6 +1246,7 @@ class TestQgsExpression: public QObject
       QTest::newRow( "format_number large" ) << "format_number(9000000.0,0)" << false << QVariant( "9,000,000" );
       QTest::newRow( "format_number many decimals" ) << "format_number(123.45600,4)" << false << QVariant( "123.4560" );
       QTest::newRow( "format_number no decimals" ) << "format_number(1999.567,0)" << false << QVariant( "2,000" );
+      QTest::newRow( "format_number language parameter" ) << "format_number(123456.789,2,'fr')" << false << QVariant( "123 456,79" );
```

Comment:
You may experience similar difficulties as https://github.com/qgis/QGIS/pull/36860 due to fr locale. Might be easiest to just use german locale instead.

Reply:
Same problem with de.